### PR TITLE
feat: add .github/pinact.yaml as a default config path

### LINKED ISSUE #393: Respect `.github/pinact.yaml` as the default config
## Feature Overview

Respect `.github/pinact.yaml` as the default config.

## Why is the feature needed?

> Please explain the problem you want to solve.

- Reduce dotfiles from the repository root.
- Place files under the directory that is used by the context.

I put the config under .github, that is clear for GitHub-specific files.
Currently, pinact needs to specify the config path to read, but this path pattern is a very standard use case like renovate.json did.

In `.github`, the dot prefix is needless. So I perfectly agree with this example path - [`.github/pinact.yaml`](https://github.com/suzuki-shunsuke/pinact/blob/baaea2b01ca9f08d28767edb4b150ef0ccbaeb35/USAGE.md?plain=1#L48).

## Example Code

> command and configuration

```console
$ pinact init '.github/pinact.yaml'
$ pinact run
```

### PR DESCRIPTION
**Pull request by kachick**
Resolves #393

### CODE REVIEW

File: pkg/controller/run/main_internal_test.go [hunk 1]
```diff
@@ -136,3 +136,49 @@ func TestController_patchLine(t *testing.T) {
 		})
 	}
 }
+
+func TestController_getConfigPath(t *testing.T) {
```
Comment (kachick) [inline pkg/controller/run/main_internal_test.go:140]: Should I add test for `readConfig` instead of this extracted `getConfigPath` function?

**Comment by kachick** [comment]
https://github.com/suzuki-shunsuke/pinact/actions/runs/8901267078/job/24444711917?pr=394

E2E diff check is failed, but `go run ./cmd/pinact run` in local made correct diff

Looks like facing API limit in GitHub action

```
time="2024-04-30T20:36:44Z" level=warning msg="get a reference" error="GET https://api.github.com/repos/suzuki-shunsuke/repo-404/commits/83b7061638ee4956cf7545a6f7efe594e5ad0247: 404 Not Found []" pinact_version= program=pinact workflow_file=testdata/foo.yaml
time="2024-04-30T20:36:44Z" level=warning msg="get a reference" error="GET https://api.github.com/repos/suzuki-shunsuke/actionlint-workflow/commits/v0.[5](https://github.com/suzuki-shunsuke/pinact/actions/runs/8901267078/job/24444711917?pr=394#step:9:6).0: 403 API rate limit of 60 still exceeded until 2024-04-30 21:03:19 +0000 UTC, not making remote request. [rate reset in 26m34s]" pinact_version= program=pinact workflow_file=testdata/foo.yaml
```

**Comment by suzuki-shunsuke** [comment]
Thank you!

**Comment by suzuki-shunsuke** [comment]
v0.1.3 is out 🎉
https://github.com/suzuki-shunsuke/pinact/releases/tag/v0.1.3

**Pull request merged**